Extensions/Video/JsExtension.js

```diff
@@ -0,0 +1,775 @@
+      .addParameter("yesorno", _("Mute or not"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("mute");
+
+    object
+      .addAction(
+        "SetTime",
+        _("Set time (in seconds)"),
+        _("Set time (in seconds)"),
+        _("Set time of my object : _PARAM0_ is _PARAM1_ _PARAM2_ seconds"),
+        "",
+        "JsPlatform/Extensions/videoicon24.png",
+        "JsPlatform/Extensions/videoicon16.png"
+      )
+      .addParameter("object", _("Choose a video object"), "", false)
+      .addParameter("operator", _("Modification's sign"), "", false)
+      .addParameter("expression", _("Time in seconds"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("setCurrentTime")
+      .setManipulatedType("number")
+      .setGetter("getCurrentTime");
+
+    object
+      .addAction(
+        "SetVolume",
+        _("Set volume (in %)"),
+        _("Set volume (in %)"),
+        _("Do _PARAM1_ _PARAM2_ % to the volume of _PARAM0_"),
+        "",
+        "JsPlatform/Extensions/videoicon24.png",
+        "JsPlatform/Extensions/videoicon16.png"
+      )
+      .addParameter("object", _("Choose a video object"), "", false)
+      .addParameter("operator", _("Modification's sign"), "", false)
+      .addParameter("expression", _("Volume in %"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("setVolume")
+      .setManipulatedType("number")
+      .setGetter("getVolume");
+
+    object
+      .addExpression(
+        "GetVolume",
+        _("Get the volume"),
+        _("Get the volume of an video object."),
+        _("Volume"),
+        "res/physics16.png"
+      )
+      .addParameter("object", _("Object"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("getVolume");
+
+    object
+      .addExpression(
+        "IsMuted",
+        _("Video is muted"),
+        _("Return if video is muted"),
+        _("Volume"),
+        "res/physics16.png"
+      )
+      .addParameter("object", _("Object"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("isMuted");
+
+    object
+      .addCondition(
+        "Play",
+        _("Is played"),
+        _("Test if an video is played"),
+        _("_PARAM0_  is played"),
+        "Control time",
+        "JsPlatform/Extensions/videoicon24.png",
+        "JsPlatform/Extensions/videoicon16.png"
+      )
+      .addParameter("object", _("Choose a video object"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("isPlayed");
+
+    object
+      .addCondition(
+        "Pause",
+        _("Is paused"),
+        _("Test if an video is paused"),
+        _("_PARAM0_  is paused"),
+        "Control time",
+        "JsPlatform/Extensions/videoicon24.png",
+        "JsPlatform/Extensions/videoicon16.png"
+      )
+      .addParameter("object", _("Choose a video object"), "", false)
+      .getCodeExtraInformation()
+      .setFunctionName("isPaused");
+
+    object
+      .addExpression(
```

Comment:
Unfortunately, booleans expressions are not yet supported in GD. This might seems to work because of JavaScript loose typinh but it's a bad idea to return a boolean while a number is expected. Could create bugs in the game and really strange behaviors => remove expressions for IsPaused, IsPlayed, IsLooped

Reply:
So i need to delete **isMuted** , **isEnded** too
It's annoying to waste these expressions 😕

Reply:
Yup, it's very tempting to do, I did something similar for the skeleton extension, a "Playing" expression, but after some meditation I removed it.

Think that they wouldn't be used too much anyway, what would they be useful to?, change an animation if the video is muted?
`Do = VideoObject.Muted() to the animation of VolumeSprite`
You can do that in two events (if muted set animation 0, if not muted 1), a bit longer but at the end improves readability and doesn't use Javascript workarounds :)

Reply:
Yeah as @Lizard-13 mentioned conditions are here for such use case :)